cmd/gen/makefile/runner.go

```diff
@@ -0,0 +1,64 @@
+package makefile
+
+import (
+	"context"
+	"io"
+	"os"
+
+	"github.com/giantswarm/devctl/pkg/gen"
+	"github.com/giantswarm/devctl/pkg/gen/input/makefile"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"github.com/spf13/cobra"
+)
+
+type runner struct {
+	flag   *flag
+	logger micrologger.Logger
+	stdout io.Writer
+	stderr io.Writer
+}
+
+func (r *runner) Run(cmd *cobra.Command, args []string) error {
+	ctx := context.Background()
+
+	err := r.flag.Validate()
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	err = r.run(ctx, cmd, args)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	return nil
+}
+
+func (r *runner) run(ctx context.Context, cmd *cobra.Command, args []string) error {
+	c := makefile.Config(*r.flag)
+
+	makefileInput, err := makefile.New(c)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	err = makefileInput.Params(ctx)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+	_, err = os.Create("Makefile")
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	err = gen.Execute(
```

Comment:
@kopiczko I moved the logic to your suggested place. How can I avoid getting the `zz_generated.Makefile` file and get a `Makefile` file? TBH I underestimated the complexity of this logic and I'm not sure how to make this right 🙂 . I would appreciate your input.

Reply:
How about that https://github.com/giantswarm/devctl/pull/57 ?